src/Symfony/Component/Validator/Tests/Constraints/UniqueValidatorTest.php

```diff
@@ -220,6 +221,64 @@ public function testExpectsValidCaseInsensitiveComparison()
 
         $this->assertNoViolation();
     }
+
+    public function testCollectionFieldsAreOptional()
+    {
+        $this->validator->validate([['value' => 5], ['id' => 1, 'value' => 6]], new Unique('id'));
+
+        $this->assertNoViolation();
+    }
+
+    /**
+     * @dataProvider getInvalidFieldNames
+     */
+    public function testCollectionFieldNamesMustBeString(string $type, mixed $field)
+    {
+        $this->expectException(UnexpectedTypeException::class);
+        $this->expectExceptionMessage(sprintf('Expected argument of type "string", "%s" given', $type));
+
+        $this->validator->validate([['value' => 5], ['id' => 1, 'value' => 6]], new Unique([$field]));
+    }
+
+    public function getInvalidFieldNames(): \Generator
+    {
+        return [
+            yield ['stdClass', new \stdClass()],
+            yield ['int', 2],
+            yield ['bool', false],
+        ];
+    }
+
+    /**
+     * @dataProvider getInvalidCollectionValues
+     */
+    public function testInvalidCollectionValues(array $value, array $fields)
+    {
+        $this->validator->validate($value, new Unique($fields, [
+            'message' => 'myMessage',
+        ]));
+
+        $this->buildViolation('myMessage')
+            ->setParameter('{{ value }}', 'array')
+            ->setCode(Unique::IS_NOT_UNIQUE)
+            ->assertRaised();
+    }
+
+    public function getInvalidCollectionValues(): \Generator
+    {
+        return [
```

Comment:
this should be removed, mixing return [...] and yield doesn't make sens, it's one style or the other :)

Reply:
Fixed. I have also refactored not mine test from which I copy-paste this approach.